### PR TITLE
Fix dead links, add CI check

### PR DESCRIPTION
**Pull request by Nutomic**
This is only covering English because I plan to handle translations differently in the future, see https://github.com/LemmyNet/lemmy-docs/issues/171.

Links in code blocks are not checked unfortunately https://github.com/becheran/mlc/issues/73

### CODE REVIEW

File: src/en/federation/overview.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Federation Overview
 
-This document is for anyone who wants to know how Lemmy federation works, without being overly technical. It is meant provide a high-level overview of ActivityPub federation in Lemmy. If you are implementing ActivityPub yourself and want to be compatible with Lemmy, read our [ActivityPub API outline](contributing_apub_api_outline.md).
+This document is for anyone who wants to know how Lemmy federation works, without being overly technical. It is meant provide a high-level overview of ActivityPub federation in Lemmy. If you are implementing ActivityPub yourself and want to be compatible with Lemmy, read our [ActivityPub Protocol](lemmy_protocol.md).
```
Comment (dessalines) [inline src/en/federation/overview.md:3]: I hope these html -> md links are working correctly, after the docs are built.

File: rust-toolchain.toml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[toolchain]
+channel = "1.61"
```
Comment (dessalines) [inline rust-toolchain.toml:2]: Why add this one?

**Review by dessalines — APPROVED** [review]

File: rust-toolchain.toml (hunk 2 above)
Comment (Nutomic) [reply rust-toolchain.toml:2]: Its failing to compile with latest cargo. Another reason for #171.

```

Compiling mdbook v0.4.15 (/home/felix/.cargo/git/checkouts/mdbook-66041be698320ee6/e74fdb1)
error[E0597]: `local_ctx` does not live long enough
--> src/renderer/html_handlebars/helpers/navigation.rs:154:25
|
154 |             t.render(r, &local_ctx, &mut local_rc, out)
|                         ^^^^^^^^^^ borrowed value does not live long enough
155 |         })?;
|         -
|         |
|         `local_ctx` dropped here while still borrowed
|         borrow might be used here, when `local_rc` is dropped and runs the destructor for type `handlebars::RenderContext<'_, '_>`
|
= note: values in a scope are dropped in the opposite order they are defined

For more information about this error, try `rustc --explain E0597`.
```

File: src/en/federation/overview.md (hunk 1 above)
Comment (Nutomic) [reply src/en/federation/overview.md:3]: Yes I tested it locally.

**Pull request merged**